src/messaging/utils/config.ts

```diff
@@ -3,45 +3,90 @@ import { TAuthConfig } from '../types';
-    await storage()
-        .set({
-            bucket: Constants.BUCKET_AUTH_STORE,
-            key: Constants.CLIENT_CREDENTIALS_BUCKET_KEY,
-            data: JSON.stringify(authConfig),
-        })
-        .catch(() => {
-            throw new StorageError(Constants.ERRORS.STORAGE_ERROR_SET);
-        });
+export const setAuthConfig = async (
+    authConfig: TAuthConfig
+): Promise<boolean> => {
+    try {
+        await storage()
+            .set({
+                bucket: Constants.BUCKET_AUTH_STORE,
+                key: Constants.BUCKET_KEY_CLIENT_CREDENTIALS,
+                data: JSON.stringify(authConfig),
+            })
+            .catch(() => {
+                throw new StorageError(Constants.ERRORS.STORAGE_ERROR_SET);
+            });
+        return true;
+    } catch (error) {
+        return false;
+    }
 };
 
-export const getConfig = async (): Promise<string> => {
-    return await storage()
-        .get({
+export const getAuthConfig = async (): Promise<TAuthConfig> => {
+    try {
+        const data = await storage().get({
             bucket: Constants.BUCKET_AUTH_STORE,
-            key: Constants.CLIENT_CREDENTIALS_BUCKET_KEY,
-        })
-        .catch(() => {
-            throw new StorageError(Constants.ERRORS.STORAGE_ERROR_GET);
+            key: Constants.BUCKET_KEY_CLIENT_CREDENTIALS,
         });
+        return JSON.parse(data);
+    } catch (error) {
+        throw new StorageError(Constants.ERRORS.STORAGE_ERROR_SET);
+    }
 };
 
-export const setAuthToken = async (authToken: string): Promise<void> => {
-    await storage().set({
-        bucket: Constants.BUCKET_AUTH_STORE,
-        key: Constants.ACCESS_TOKEN_BUCKET_KEY,
-        data: JSON.stringify(authToken),
-    });
+export const setAuthToken = async (authToken: string): Promise<boolean> => {
+    try {
+        await storage().set({
+            bucket: Constants.BUCKET_AUTH_STORE,
+            key: Constants.BUCKET_KEY_ACCESS_TOKEN,
+            data: JSON.stringify(authToken),
+        });
+        return true;
+    } catch (error) {
+        return false;
+    }
 };
 
 export const getAuthToken = async (): Promise<string | boolean> => {
     try {
         const data = await storage().get({
             bucket: Constants.BUCKET_AUTH_STORE,
-            key: Constants.ACCESS_TOKEN_BUCKET_KEY,
+            key: Constants.BUCKET_KEY_ACCESS_TOKEN,
         });
         return JSON.parse(data);
     } catch (error) {
         return false;
     }
 };
+
+export const setAuthTokenRetryCount = async (
+    retryCount: string
+): Promise<boolean> => {
+    try {
+        await storage().set({
+            bucket: Constants.BUCKET_AUTH_STORE,
+            key: Constants.BUCKET_KEY_AUTH_RETRY_COUNT,
+            data: JSON.stringify(retryCount),
+        });
+        return true;
+    } catch (error) {
+        return false;
+    }
+};
+
+export const getAuthTokenRetryCount = async (): Promise<number> => {
+    try {
```

Comment:
could this ever end up in an infinite retry loop?

Reply:
Good pickup, if a clients storage fails to store it could loop here.

Will change to something like the following.

`export const getAuthTokenRetryCount = async (): Promise<number> => {
    try {
        const data = await storage().get({
            bucket: Constants.BUCKET_AUTH_STORE,
            key: Constants.BUCKET_KEY_AUTH_RETRY_COUNT,
        });
        return parseInt(JSON.parse(data));
    } catch (error) {
        return 0;
    }
};
`

Returns the following on an expired token.

`
        this.authRetryCount = await getAuthTokenRetryCount();
        console.debug('DEBUG:authRetryCount:', this.authRetryCount);
`

<img width="1075" alt="Screen Shot 2021-07-06 at 10 27 41 am" src="https://user-images.githubusercontent.com/1700678/124526027-cccc1000-de44-11eb-8677-135e3dfcb1f3.png">

Reply:
pushed updates